src/main/drivers/rx_nrf24l01.c

```diff
@@ -0,0 +1,346 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// This file is copied with modifications from project Deviation,
+// see http://deviationtx.com
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <platform.h>
+
+#ifdef USE_RX_NRF24
+
+#include "system.h"
+#include "gpio.h"
+#include "rx_nrf24l01.h"
+#ifndef UNIT_TEST
+#include "bus_spi.h"
+#endif
+
+/* Instruction Mnemonics */
+#define R_REGISTER    0x00
+#define W_REGISTER    0x20
+#define REGISTER_MASK 0x1F
+#define ACTIVATE      0x50
+#define R_RX_PL_WID   0x60
+#define R_RX_PAYLOAD  0x61
+#define W_TX_PAYLOAD  0xA0
+#define W_ACK_PAYLOAD 0xA8
+#define FLUSH_TX      0xE1
+#define FLUSH_RX      0xE2
+#define REUSE_TX_PL   0xE3
+#define NOP           0xFF
+
+#if defined(UNIT_TEST)
+#define DISABLE_NRF24() {}
+#define ENABLE_NRF24() {}
+
+#define NRF24_CE_HI() {}
+#define NRF24_CE_LO() {}
+#else
+#define DISABLE_NRF24()     {GPIO_SetBits(NRF24_CS_GPIO, NRF24_CS_PIN);}
```

Comment:
CF has lots of legacy code/coding style. It would be nice to phase it out eventually ...